rest-api/src/main/java/com/cloudbees/workflow/rest/external/RunExt.java

```diff
@@ -332,8 +332,8 @@ public static RunExt createOld(WorkflowRun run) {
             if (artifacts != null && !artifacts.isEmpty()) {
                 runExt.get_links().setArtifacts(Link.newLink(RunAPI.getArtifactsUrl(run)));
             }
-
             FlowGraphWalker walker = new FlowGraphWalker(execution);
+            List<FlowNode> sortedNodes = FlowNodeUtil.getIdSortedExecutionNodeList(execution); // Hold a ref to prevent GC until we're done analyzing
```

Comment:
I'm not convinced this would actually prevent it from being removed from the cache.